### PR TITLE
Added lecture 25  time series

### PR DESCRIPTION
**Pull request by vinaykale64**

### CODE REVIEW

**Comment by amueller** [comment]
~~this will actually be lecture 26, can you rename it?~~

**Comment by amueller** [comment]
Never mind, off by one error.... all good.

File: slides/aml-25-042518-time-series/index.html [hunk 1]
```diff
@@ -0,0 +1,952 @@
...
+???
+
+---
+
+# Parse Dates
+
+.smaller[
+```python
+maunaloa = pd.read_csv(url, skiprows=49, header=None, delim_whitespace=True,
+                       names=names, parse_dates=[[0, 1, 2]], na_values=[-999.99])
```
Comment (amueller) [inline slides/aml-25-042518-time-series/index.html:260]: if you remove a note from the slide like here, an you please add it to the slide notes?

File: slides/aml-25-042518-time-series/index.html [hunk 2]
```diff
@@ -0,0 +1,952 @@
...
+```python
+maunaloa = pd.read_csv(url, skiprows=49, header=None, delim_whitespace=True,
+                       names=names, parse_dates=[[0, 1, 2]], na_values=[-999.99],
+                       index_col="year_month_day")
+maunaloa.head()                       
+```
+]
+
+.center[![:scale 80%](images/time series index.png)]
+
```
Comment (amueller) [inline slides/aml-25-042518-time-series/index.html:284]: note missing "Can also use set_index if we already read the data it."

File: slides/aml-25-042518-time-series/index.html [hunk 3]
```diff
@@ -0,0 +1,952 @@
...
+year_month_day
+1974-05-19    333.34
+1974-05-26    332.95
+1974-06-02    332.32
+1974-06-09    332.18
+1974-06-16    332.37
+Name: co2, dtype: float64
+```
+```python
+maunaloa.head()
```
Comment (amueller) [inline slides/aml-25-042518-time-series/index.html]: should be maunaloa.co2.plot but you can also just remove it.

File: slides/aml-25-042518-time-series/index.html [hunk 4]
```diff
@@ -0,0 +1,952 @@
...
+]
+]
+
+.right-column[
+
+.center[![:scale 80%](images/resampling 1.png)]
+.center[![:scale 80%](images/resampling 2.png)]
+
+]
+
```
Comment (amueller) [inline slides/aml-25-042518-time-series/index.html]: link missing http://pandas.pydata.org/pandas-docs/stable/timeseries.html#timeseries-offset-aliases

File: slides/aml-25-042518-time-series/index.html [hunk 5]
```diff
@@ -0,0 +1,952 @@
...
+???
+
+---
+
+# De-trending
+
+- Model trend
+- Or differencing (compute new series)....  $\quad x_{t + 1} - x_t$
+
```
Comment (amueller) [inline slides/aml-25-042518-time-series/index.html:430]: ``ppm.diff.plot()`` missing.

File: slides/aml-25-042518-time-series/index.html [hunk 6]
```diff
@@ -0,0 +1,952 @@
...
+.right-column[
+
+- More complicated process model
+- Internally uses differencing
+- Also checkout SARIMAX (state space model)
+
+]
+
+???
+
```
Comment (amueller) [inline slides/aml-25-042518-time-series/index.html:654]: can you make a note about specifying type='levels' in predict? (was highlighted on the slide)

File: slides/aml-25-042518-time-series/index.html [hunk 7]
```diff
@@ -0,0 +1,952 @@
...
+---
+
+# Quadratic Model
+
+.smaller[
+```python
+from sklearn.preprocessing import PolynomialFeatures
+lr_poly = make_pipeline(PolynomialFeatures(include_bias=False), 
+                          LinearRegression())
```
Comment (amueller) [inline slides/aml-25-042518-time-series/index.html:705]: indentation

File: slides/aml-25-042518-time-series/index.html [hunk 8]
```diff
@@ -0,0 +1,952 @@
...
+
+???
+FIXME - Note: Couldn't decrease size further
+
+
+---
+
+# Just Linear Regression
+
+.smaller[
```
Comment (amueller) [inline slides/aml-25-042518-time-series/index.html:781]: can you use tiny-code or smallest here to make it fit?

File: slides/aml-25-042518-time-series/index.html [hunk 9]
```diff
@@ -0,0 +1,952 @@
...
+
+---
+
+.center[![:scale 90%](images/arma month.png)]
+
+???
+
+---
+
+.smaller[
```
Comment (amueller) [inline slides/aml-25-042518-time-series/index.html:811]: smaller again?

File: slides/aml-25-042518-time-series/index.html [hunk 10]
```diff
@@ -0,0 +1,952 @@
...
+---
+
+# Multivariate Timeseries
+
+.smaller[
+- energy data from https://archive.ics.uci.edu/ml/datasets/Appliances+energy+prediction
+
+```python
+energy = pd.read_csv("energydata_complete.csv", parse_dates=['date'], 
+                      index_col="date")
```
Comment (amueller) [inline slides/aml-25-042518-time-series/index.html:850]: indentation

File: slides/aml-25-042518-time-series/index.html [hunk 11]
```diff
@@ -0,0 +1,952 @@
...
+
+???
+
+---
+
+# Weekly trends
+
+.smaller[
+```python
+week = energy.Appliances.groupby([energy.index.dayofweek, energy.index.hour]).mean()
```
Comment (amueller) [inline slides/aml-25-042518-time-series/index.html]: either decrease font-size or break line (into two calls, say a second call just doing mean)

**Review by amueller — COMMENTED** [review]
great thanks. some nitpicks.

**Comment by amueller** [comment]
Do you think you'll have time to address these this morning?

**Comment by vinaykale64** [comment]
Dear Professor

Yes, I will try to update these. Apologies for the delay, I had a final yesterday night , so was occupied with it.

**Comment by amueller** [comment]
Thanks and no worries.

File: slides/aml-25-042518-time-series/index.html (hunk 1 above)
Comment (vinaykale64) [reply slides/aml-25-042518-time-series/index.html:260]: done

File: slides/aml-25-042518-time-series/index.html (hunk 2 above)
Comment (vinaykale64) [reply slides/aml-25-042518-time-series/index.html:284]: done

File: slides/aml-25-042518-time-series/index.html (hunk 3 above)
Comment (vinaykale64) [reply slides/aml-25-042518-time-series/index.html]: done

File: slides/aml-25-042518-time-series/index.html (hunk 4 above)
Comment (vinaykale64) [reply slides/aml-25-042518-time-series/index.html]: done

File: slides/aml-25-042518-time-series/index.html (hunk 5 above)
Comment (vinaykale64) [reply slides/aml-25-042518-time-series/index.html:430]: done

File: slides/aml-25-042518-time-series/index.html (hunk 6 above)
Comment (vinaykale64) [reply slides/aml-25-042518-time-series/index.html:654]: done

File: slides/aml-25-042518-time-series/index.html (hunk 7 above)
Comment (vinaykale64) [reply slides/aml-25-042518-time-series/index.html:705]: I didn't get what was the error here

File: slides/aml-25-042518-time-series/index.html (hunk 1 above)
Comment (vinaykale64) [reply slides/aml-25-042518-time-series/index.html:260]: Actually most of the missing parts are due to me referring to the .odp file and not the notes.pdf

File: slides/aml-25-042518-time-series/index.html (hunk 8 above)
Comment (vinaykale64) [reply slides/aml-25-042518-time-series/index.html:781]: done

File: slides/aml-25-042518-time-series/index.html (hunk 1 above)
Comment (amueller) [reply slides/aml-25-042518-time-series/index.html:260]: I was looking at the odp file. Opening it in power point must have swallowed it.

File: slides/aml-25-042518-time-series/index.html (hunk 7 above)
Comment (amueller) [reply slides/aml-25-042518-time-series/index.html:705]: LinearRegression should be indented two spaces less. Don't worry about it.

File: slides/aml-25-042518-time-series/index.html (hunk 9 above)
Comment (vinaykale64) [reply slides/aml-25-042518-time-series/index.html:811]: done

File: slides/aml-25-042518-time-series/index.html (hunk 10 above)
Comment (vinaykale64) [reply slides/aml-25-042518-time-series/index.html:850]: didn't get the error

Comment (amueller) [reply slides/aml-25-042518-time-series/index.html:850]: ``index_col`` is indented one space too much. I can fix it.

File: slides/aml-25-042518-time-series/index.html (hunk 11 above)
Comment (vinaykale64) [reply slides/aml-25-042518-time-series/index.html]: done

**Comment by amueller** [comment]
thanks.

**Pull request merged**